### ISSUE TITLE
http_client suppresses https

### ISSUE THREAD

**Issue by haasted** [tendermint/tendermint#4148]
**Tendermint version**: v0.32.7

**What happened**:
The file `rpc/lib/client/http_client.go` appears to suppress usage of the `https` protocol in endpoints.

1. The [`toClientAddrAndParse`](https://github.com/tendermint/tendermint/blob/v0.32.7/rpc/lib/client/http_client.go#L40) function accepts an endpoint url.

2. It decomposes the URL into protocol and address part using the [`parseRemoteAddr`](https://github.com/tendermint/tendermint/blob/v0.32.7/rpc/lib/client/http_client.go#L72) function.

3. `parseRemoteAddr` translates protocols `http`and `https` into protocol `tcp` in line 86-89.

4. Back in `toClientAddrAndParse`, any "unknown protocols" (which includes tcp) defaults to http. 

In short, my attempt to call `https://lilmermaid.validator.network:443` gets transformed into `http://lilmermaid.validator.network:443` by the above code. 

**What you expected to happen**:

`https` should be considered a valid (perhaps even preferred) protocol for RPC endpoints.

**Have you tried the latest version**: yes

**How to reproduce it**: Attempted to describe this above.

**Comment by marbar3778** [comment]
I believe this is a duplicate of https://github.com/tendermint/tendermint/issues/4051 fixed in master

**Comment by haasted** [comment]
@marbar3778 : I concur.

**Comment by marbar3778** [comment]
will close this issue then 😄
